### PR TITLE
Add couchdb image for integration testing

### PR DESCRIPTION
**Pull request by therve**

### CODE REVIEW

**Comment by therve** [comment]
For info the image we used (klaemo/couchdb) got deprecated and removed, and there is no published image that provide a development cluster with a released version. This achieves that.

File: couch/2.3/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+ && cd couchdb \
+ # Build the release and install into /opt
+ && ./configure \
+ && make all
+
+# Setup directories and permissions
+RUN chown -R couchdb:couchdb /usr/src/couchdb
+
+WORKDIR /opt/couchdb
+EXPOSE 5984 15984 25984 35984
```
Comment (ofek) [inline couch/2.3/Dockerfile]: Need all?

File: couch/2.3/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    libicu57 \
+    libmozjs185-1.0 \
+    openssl && \
+  rm -rf /var/lib/apt/lists/*
+
+# grab gosu for easy step-down from root and tini for signal handling
+# see https://github.com/apache/couchdb-docker/pull/28#discussion_r141112407
+ENV GOSU_VERSION 1.10
+ENV TINI_VERSION 0.16.1
+RUN set -ex; \
```
Comment (ofek) [inline couch/2.3/Dockerfile]: Let's get rid of -ex and use the more idiomatic `&&` instead of `;`

File: couch/2.3/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    python3-sphinx
+
+RUN pip3 install --upgrade \
+    sphinx_rtd_theme \
+    nose \
+    requests \
+    hypothesis
+
+# Node is special
+RUN set -ex; \
```
Comment (ofek) [inline couch/2.3/Dockerfile]: Let's get rid of -ex and use the more idiomatic `&&` instead of `;`

File: couch/2.3/Dockerfile (hunk 2 above)
Comment (therve) [reply couch/2.3/Dockerfile]: Done

File: couch/2.3/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    erlang-nox \
+    erlang-reltool \
+    erlang-dev \
+    libcurl4-openssl-dev \
+    libicu-dev \
+    libmozjs185-dev \
+    python3 \
+    libpython3-dev \
+    python3-pip \
+    python3-sphinx
```
[the remark this answers is not in the document]
Comment (therve) [reply couch/2.3/Dockerfile]: I did it in the last apt call.

File: couch/2.3/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+RUN pip3 install --upgrade \
+    sphinx_rtd_theme \
+    nose \
+    requests \
+    hypothesis
```
[the remark this answers is not in the document]
Comment (therve) [reply couch/2.3/Dockerfile]: Good point, it turns out we don't need any of that.

File: couch/2.3/Dockerfile (hunk 3 above)
Comment (therve) [reply couch/2.3/Dockerfile]: Done

File: couch/2.3/Dockerfile (hunk 1 above)
Comment (therve) [reply couch/2.3/Dockerfile]: Just the first.

File: couch/2.3/Dockerfile (hunk 4 above)
[the remark this answers is not in the document]
Comment (ofek) [reply couch/2.3/Dockerfile]: If you don't do it here too then subsequent layers will increase in size

**Pull request merged**